cpp/src/arrow/compute/kernels/scalar_compare_test.cc

```diff
@@ -1865,5 +1861,636 @@ TEST(TestMaxElementWiseMinElementWise, CommonTemporal) {
+      ArrayFromJSON(TypeTraits<StringType>::type_singleton(), R"(["z","a","g"])"),
+      ArrayFromJSON(TypeTraits<BooleanType>::type_singleton(), "[true, true, true]"));
+  ValidateBetween<StringType>(
+      InclusiveOption,
+      ArrayFromJSON(TypeTraits<StringType>::type_singleton(),
+                    R"(["block","bit","binary"])"),
+      ArrayFromJSON(TypeTraits<StringType>::type_singleton(),
+                    R"(["bit","nibble","ternary"])"),
+      ArrayFromJSON(TypeTraits<StringType>::type_singleton(), R"(["word","d","xyz"])"),
+      ArrayFromJSON(TypeTraits<BooleanType>::type_singleton(), "[true, false, false]"));
+  ValidateBetween<StringType>(
+      InclusiveOption,
+      ArrayFromJSON(TypeTraits<StringType>::type_singleton(),
+                    R"(["よしもと","の","ち"])"),
+      ArrayFromJSON(TypeTraits<StringType>::type_singleton(), R"(["は","へ","あ"])"),
+      ArrayFromJSON(TypeTraits<StringType>::type_singleton(), R"(["な","を","ち"])"),
+      ArrayFromJSON(TypeTraits<BooleanType>::type_singleton(), "[false, false, true]"));
+  ValidateBetween<StringType>(
+      InclusiveOption,
+      ArrayFromJSON(TypeTraits<StringType>::type_singleton(), R"(["A","ア","王"])"),
+      ArrayFromJSON(TypeTraits<StringType>::type_singleton(), R"(["た","あ","歩"])"),
+      ArrayFromJSON(TypeTraits<StringType>::type_singleton(), R"(["李","田",null])"),
+      ArrayFromJSON(TypeTraits<BooleanType>::type_singleton(), "[false, true, null]"));
+  ValidateBetween<StringType>(
+      InclusiveOption,
+      ArrayFromJSON(TypeTraits<StringType>::type_singleton(), R"(["Б",null,"Я"])"),
+      ArrayFromJSON(TypeTraits<StringType>::type_singleton(), R"(["А","Ж","Щ"])"),
+      ArrayFromJSON(TypeTraits<StringType>::type_singleton(), R"(["Д","Л","Ф"])"),
+      ArrayFromJSON(TypeTraits<BooleanType>::type_singleton(), "[true, null, false]"));
+}
+
+TEST(TestTimestampsBetweenKernel, TimestampsArrayArrayArrayTest) {
+  const std::string arr_json = R"(["1970-01-01","2000-02-02","1900-02-28"])";
+  const std::string lhs_json = R"(["1970-01-01","2000-02-01","1900-02-28"])";
+  const std::string rhs_json = R"(["1970-01-02","2000-02-02","1900-02-28"])";
+  // Between Options
+  BetweenOptions both(BetweenOptions::Inclusive::BOTH);
+  BetweenOptions left(BetweenOptions::Inclusive::LEFT);
+  BetweenOptions right(BetweenOptions::Inclusive::RIGHT);
+  BetweenOptions neither(BetweenOptions::Inclusive::NEITHER);
+  // Same units should be fine
+  auto arr = ArrayFromJSON(timestamp(TimeUnit::SECOND), arr_json);
+  auto lhs = ArrayFromJSON(timestamp(TimeUnit::SECOND), lhs_json);
+  auto rhs = ArrayFromJSON(timestamp(TimeUnit::SECOND), rhs_json);
+  // BOTH
+  ValidateBetween<TimestampType>(both, arr, lhs, rhs);
+  // LEFT
+  ValidateBetween<TimestampType>(left, arr, lhs, rhs);
+  // RIGHT
+  ValidateBetween<TimestampType>(right, arr, lhs, rhs);
+  // NEITHER
+  ValidateBetween<TimestampType>(neither, arr, lhs, rhs);
```

Comment:
Nit: No need for these comments as the code already is clear enough.

Reply:
Removed.